Update references to the (renamed) main branch

Neverlord opened the pull request:
We have renamed our development branch to `main`. This PR simply updates all references to the branch in CI script and in our documentation.

riemass commented:
There is a reference to master in the `README.md` file:

> Our CI covers Windows (latest MSVC release), macOS (latest Xcode release),
> FreeBSD (latest) and several Linux distributions (via the Docker images found
> \[here\]\(https://github.com/actor-framework/actor-framework/tree/master/.ci\)). 

Also, just as a checkmark, do you think any changes are needed to the [continuous integration project](https://github.com/actor-framework/continuous-integration)?

Neverlord commented:
> There is a reference to master in the `README.md` file (...)

Good catch, thanks! Fixed.

> Also, just as a checkmark, do you think any changes are needed to the [continuous integration project](https://github.com/actor-framework/continuous-integration)?

I've already updated the Jenkins library: https://github.com/actor-framework/continuous-integration/commit/709eb307ebb9943238290efdd531b41f400c94b0
